### PR TITLE
feat(EVM/Fantom): CW-7854 Display Stake on CWP when Signing Staking Smart Contract

### PR DESCRIPTION
**Pull request by kanitsu**

### CODE REVIEW

**Comment by cbx-tech** [comment]
Task linked: [CW-7854 SDK update stake display on CWP](https://app.clickup.com/t/25577573/CW-7854)

File: packages/coin-evm/src/index.ts [hunk 1]
```diff
@@ -93,7 +94,11 @@ class Evm extends COIN.ECDSACoin {
       );
     }
 
-    const script = SCRIPTS.signSmartContractTransaction.scriptWithSignature;
+    let script = SCRIPTS.signSmartContractTransaction.scriptWithSignature;
+    // Staking contract for Fantom
+    if(this.chain.id === FANTOM.id && client.transaction.to.toLowerCase() === FANTOM.stakingContract.toLowerCase()){
```
Comment (skchang0710) [inline packages/coin-evm/src/index.ts:99]: The staking contract might apply different methods. Should the condition be more precise for corresponding signing scripts ?

Comment (kanitsu) [reply packages/coin-evm/src/index.ts:99]: You mean checking data for program id, and allowing only certain known program, for example: delegation (0x9fa6dd35) and withdraw (0x0962ef79)?

Comment (skchang0710) [reply packages/coin-evm/src/index.ts:99]: Yes, we are used to handle these staking methods by showing method names on screen. You could refer to the procedure of ETH smart contracts in the SDK, or you could ask PO for proper implementation in the task.

Comment (kanitsu) [reply packages/coin-evm/src/index.ts:100]: Staking detection is more precise and the display is updated according to PO request.

File: packages/coin-evm/src/utils/arguments.ts [hunk 2]
```diff
@@ -51,6 +52,34 @@ async function getSELegacySmartContractTransaction(
...
+  const programId = transaction.data.slice(0, 10);
+  let stakingCode = '';
+  if (programId.toLowerCase() === FANTOM.delegateProgram) {
+    stakingCode = '01';
+  } else if (programId.toLowerCase() === FANTOM.withdrawProgram) {
+    stakingCode = '02';
+  } else if (programId.toLowerCase() === FANTOM.undelegateProgram) {
+    stakingCode = '03';
+  } else {
+    throw new error.SDKError(getSELegacyStakingTransaction.name, 'Unrecognized program id for staking contract.');
```
Comment (skchang0710) [inline packages/coin-evm/src/utils/arguments.ts:75]: What if the other staking methods are required ?

Comment (kanitsu) [reply packages/coin-evm/src/utils/arguments.ts:75]: Then we update SDK and SCRIPT. The alternative would be allowing all program without recognizing it, i.e. display only [STAKE], which in my opinion a little bit unsafe. Do you have better alternative? I'm open to suggestions.

**Comment by kanitsu** [comment]
Merge staking script to legacy smart script.
Add type argument to differentiate smart contract and staking transactions.
Unknown staking type will display "SMART".
Delegate and Undelegate will also display amount after validator ID.

**Review by hankliu418 — APPROVED** [review]

**Pull request merged**